core/src/main/java/com/digitalpebble/stormcrawler/util/MetadataTransfer.java

```diff
@@ -156,12 +156,34 @@ public Metadata filter(Metadata metadata) {
         return filtered_md;
     }
 
+    /**
+     * Filter the metadata based on a set of keys. If a key ends with a * then all the keys starting
+     * with the prefix will be added.
+     */
     private Metadata _filter(Metadata metadata, Set<String> filter) {
         Metadata filtered_md = new Metadata();
+
+        Set<String> filterKeys = new HashSet<>();
         for (String key : filter) {
-            String[] vals = metadata.getValues(key);
-            if (vals != null) filtered_md.setValues(key, vals);
+            if (key.endsWith("*")) {
+                String prefix = key.substring(0, key.length() - 1);
+                for (String mdKey : metadata.keySet()) {
+                    if (mdKey.startsWith(prefix)) {
+                        filterKeys.add(mdKey);
+                    }
+                }
+            } else {
+                filterKeys.add(key);
+            }
+
+            for (String filterKey : filterKeys) {
+                String[] values = metadata.getValues(filterKey);
```

Comment:
Could add a method _copy(Metadata target, String key, boolean allowNull)_ to Metadata. I am sure a similar operation is done elsewhere in the code and we might as well make it easier.

Instead of storing the keys in a set then iterate on it, could simply 


  ```
if (mdKey.startsWith(prefix)) {
                        metadata.copy(filtered_md, mdKey, false);
                    }
                }
            } else {
                        metadata.copy(filtered_md, key, false);
            }
```

Reply:
In the new commit, I added `copy(targetMetadata, key)`. The `allowNull` flag is actually not necessary, I found out, because in `metadata.SetValues(...)` null values are ignored anyways. This copy method takes only a single key. Also having a copy method with a set of input keys would be possible.  
I didn't find any other similar operation elsewhere in the code.